src/backend/seq/genpup.py

```diff
@@ -6,17 +6,22 @@
 
 import sys
 import argparse
+
 sys.path.append('maint/')
 import yutils
 
-indentation = 0
-num_paren_open = 0
+sys.path.append('src/backend/')
+import gencomm
+
+
+########################################################################################
+##### Global variables
+########################################################################################
 
-def display(*argv):
-    for x in range(indentation):
-        OUTFILE.write("    ")
-    for arg in argv:
-        OUTFILE.write(arg)
+num_paren_open = 0
```

Comment:
You already have `yutils.display` to track indentation based on braces, it should be easy enough to let it also track `num_parent_open`, and say, let `yutils.display(OUTFILE, "WRAP_BRACES")` to automatically wrap all open braces. In fact, aren't `num_paren_open` always the same as `display.indent`?

Reply:
In the current version, yes.  But I don't know how many parenthesis to close in some places.  I think we can leave it in for now.